lib/tasks/batches.rake

```diff
@@ -0,0 +1,15 @@
+namespace :batches do
+  desc "normalize specimen_role values in Batches to be one of the options listed in fields.yml"
+  task normalize_specimen_roles: :environment do
+    Batch.find_each do |batch|
+      unless batch.specimen_role.nil?
```

Comment:
I think we could write this block in a clearer way, like this:
```
Batch.find_each do |batch|
   next if batch.specimen_role.nil?

   specimen_role_id = batch.specimen_role[0].downcase
   batch.specimen_role = (Batch::SPECIMEN_ROLES_IDS.include? specimen_role_id) ? specimen_role_id : nil

   batch.save
end
```

Reply:
There are other alternatives to avoid issues around migrations to also migrate data. For example you can regularly replace old migrations with a single one, copying the current `db/schema.rb`  (never seen that one used), or you can migrate data using SQL and static data, which is often possible, and which I'm starting to use more and more. For example:

```sql
UPDATE batches
SET specimen_role = LOWER(LEFT(specimen_role, 1))
WHERE specimen_role REGEXP '^[bcefglopqrv]';

UPDATE batches
SET specimen_role = NULL
WHERE specimen_role NOT IN ('b', 'c', 'e', 'f', 'g', 'l', 'o', 'p', 'q', 'r', 'v');
```

A benefit is that it's incredibly faster, especially on large datasets, so migrations are smooth :boat: 
A drawback is that if the data changes before the release, we must remember to manually update the list here :imp:

Reply:
Also `Batch.where.not(specimen_role: nil).find_each` to only load batches with a specimen_role.